experiment/autobumper/main.go

```diff
@@ -0,0 +1,323 @@
+	output, err := cmd.Output()
+	if err != nil {
+		return err
+	}
+	d := strings.TrimSpace(string(output))
+	logrus.Infof("Changing working directory to %s...", d)
+	return os.Chdir(d)
+}
+
+type options struct {
+	githubLogin string
+	githubToken string
+	gitName     string
+	gitEmail    string
+}
+
+func parseOptions() options {
+	var o options
+	flag.StringVar(&o.githubLogin, "github-login", "", "The GitHub username to use.")
+	flag.StringVar(&o.githubToken, "github-token", "", "The path to the GitHub token file.")
+	flag.StringVar(&o.gitName, "git-name", "", "The name to use on the git commit. Requires --git-email. If not specified, uses the system default.")
+	flag.StringVar(&o.gitEmail, "git-email", "", "The email to use on the git commit. Requires --git-name. If not specified, uses the system default.")
+	flag.Parse()
+	return o
+}
+
+func validateOptions(o options) error {
+	if o.githubLogin == "" {
+		return fmt.Errorf("--github-login is mandatory")
+	}
+	if o.githubToken == "" {
+		return fmt.Errorf("--github-token is mandatory")
+	}
+	if (o.gitEmail == "") != (o.gitName == "") {
+		return fmt.Errorf("--git-name and --git-email must be specified together")
+	}
+	return nil
+}
+
+func call(cmd string, args ...string) error {
+	c := exec.Command(cmd, args...)
+	c.Stdout = os.Stdout
+	c.Stderr = os.Stderr
+	return c.Run()
+}
+
+func updatePR(gc *github.Client, org, repo, title, body, matchTitle, source, branch string) error {
+	logrus.Info("Creating PR...")
+	n, err := updater.UpdatePR(org, repo, title, body, matchTitle, gc)
+	if err != nil {
+		return fmt.Errorf("failed to update %d: %v", n, err)
+	}
+	if n == nil {
+		pr, err := gc.CreatePullRequest(org, repo, title, body, source, branch, true)
+		if err != nil {
+			return fmt.Errorf("failed to create PR: %v", err)
+		}
+		n = &pr
+	}
+
+	logrus.Infof("PR %s/%s#%d will merge %s into %s: %s", org, repo, *n, source, branch, title)
+	return nil
+}
+
+func updateReferences() (map[string]string, error) {
+	logrus.Info("Bumping image references...")
+	filter := regexp.MustCompile(prowPrefix + "|" + testImagePrefix)
+
+	err := filepath.Walk(".", func(path string, info os.FileInfo, err error) error {
+		if strings.HasSuffix(path, ".yaml") {
+			if err := bumper.UpdateFile(path, filter); err != nil {
+				logrus.WithError(err).Errorf("Failed to update path %s.", path)
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return bumper.GetReplacements(), nil
+}
+
+func getNewProwVersion(images map[string]string) string {
+	for k, v := range images {
+		if strings.HasPrefix(k, prowPrefix) {
+			return v
+		}
+	}
+	return ""
+}
+
+func makeCommitSummary(images map[string]string) string {
+	return fmt.Sprintf("Updated prow to %s, and other images as necessary.", getNewProwVersion(images))
+}
+
+func updateConfig() error {
+	logrus.Info("Updating generated config...")
+	return call("bazel", "run", "//hack:update-config")
```

Comment:
Maybe drop a TODO to do something different here

Reply:
Done.